pkg/resources/flow.go

```diff
@@ -87,11 +86,32 @@ func (f *Flow) Create() error {
 		}
 	}
 	options := interfaces.DependencyGraphOptions{
-		FlowName: f.originalName,
-		Args:     args,
+		FlowName:         f.originalName,
+		Args:             args,
+		FlowInstanceName: f.generatedName,
 	}
+
+	if destroy {
+		// Delete one replica
+		options.ReplicaCount = -1
+	} else {
+		// Recheck existing replica resources or create one replica if none exist
+		options.ReplicaCount = 0
+		options.MinReplicaCount = 1
+	}
+
 	graph, err := f.context.Scheduler().BuildDependencyGraph(options)
 	if err != nil {
+		return nil, err
+	}
+	return graph, nil
+
+}
+
+// Triggers the flow deployment like it was the resource creation
```

Comment:
Flow is a virtual resource. Before flows all graph nodes were resource definitions and each definition eventually became a k8s resource. Flows bring another concept - now graph nodes can be not just resources, but also some processing instructions (flow node is similar to function call in programming languages, in the future we might decide to have nodes that correspond to condition (i.e. if-then-else), or to represent operation other than create (say specify patch to existing resource)). Initially I wanted to create another TPR fro flows. But it turns out that this greatly complicates the code, because from Dependency standpoint there is no distinction between node types, it is very convenient that you can select all graph nodes with one query and then put them in single collection. Besides flows interfaces is Do/Undo which is basically the same as Create/Delete for resources. Thus I decided not to create additional TPRs and instead make flow be something that pretends to be a resource, but doesn't create anything in k8s. Instead Create method triggers flow (replica) deployment and Delete does the opposite